src/aws_s3_auth_conn.tsx

```diff
@@ -59,12 +57,15 @@ export class MyS3Auth {
   }
 
   // Check if the user's keys are valid for specified bucket
+  // Save the S3 object made upon validation
   // Code after this function call will likely execute before this function finishes
-  checkAndDisplayValidUser(fnSucceed : (args : string) => any, args : string ) {
+  checkAndDisplayValidUser(setViewState : (args : string) => any, args : string, setS3Obj : (s3Obj : MyS3Auth) => any, s3Obj : MyS3Auth ) {
+    
     const command = new HeadBucketCommand({ Bucket: this.whichBucket })
     this.s3Client.send(command).then((data) => {
-      console.log(data)
-      fnSucceed(args)
+      console.log(data);
+      setS3Obj(s3Obj);
```

Comment:
How come we only set the global variable here?

Reply:
S3 object should be global only if the object made with the user's keys are valid. So naturally it'd make sense to store it after the promise was fulfilled